### DISCUSSION TITLE
Is complex mode supported?

### DISCUSSION THREAD

**Discussion by JunmingDuan** [firedrakeproject/firedrake#2575] [Firedrake support] [answered]
Dear all,

I have searched around but found no tutorial related to the complex number.
The complex mode is mentioned in UFL document, but I don't know how to use.
For example, if I use
f = Function(V)
f.interpolate(1j*x),
it reports "ufl.log.UFLException: Unexpected complex value in real expression"
Is it possible to use complex number now?
Thanks for any help!

**Comment by colinjcotter** [comment, marked as answer]
Hi Junming,
   You need to install a separate firedrake distribution that uses complex instead of real numbers. This is done by passing —complex to the install script.

all the best
cjc

On 8 Oct 2022, at 17:39, Junming Duan ***@***.***> wrote:

﻿

Dear all,

I have searched around but found no tutorial related to the complex number.
The complex mode is mentioned in UFL document, but I don't know how to use.
For example, if I use
f = Function(V)
f.interpolate(1j*x),
it reports "ufl.log.UFLException: Unexpected complex value in real expression"
Is it possible to use complex number now?
Thanks for any help!

**Comment by rckirby** [comment]
And you might want to create a separate virtual environment through —venv-name
